File: mod/library/server/handle-message/exporter/process-suit.ts
import outputImage from "./_common/output-image.ts";

/**
 *
 * @param sets
 * @param suit
 * @noSelf
 * @example
 */
const processSuit = (suit: typeof SMODS.Suit) => {
	const item: any = {};

	const input = {
		...suit,
		atlas: suit.lc_atlas || "cards_1",
		pos: {
			...suit.pos,
			x: 12
		}
	};

	outputImage(input, "suit_");

	item.name = G.localization.misc.suits_plural[suit.key];
	item.key = suit.key;
	if (suit.mod !== undefined && suit.mod.id !== "Aura" && suit.mod.id !== "aure_spectral") {
		item.mod = suit.mod.id;
	}
	if (item.name) {
		return item;
	}
};

export default processSuit;

File: mod/library/server/handle-message/exporter/process_suit.lua
local function process_suit(sets, suit)
	local item = {}
	item.name = G.localization.misc.suits_plural[suit.key]
	item.key = suit.key
	if suit.mod and suit.mod.id ~= "Aura" and suit.mod.id ~= "aure_spectral" then
		item.mod = suit.mod.id
	end
	if item.name then
		sets["Suit"][item.key] = item
	end
end

return process_suit
